extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -175,14 +182,13 @@ public void buildReactivePersistenceUnit(
 
         // We only support Hibernate Reactive with a reactive data source, otherwise we don't configure the PU
         DataSourcesReactiveBuildTimeConfig.DataSourceReactiveOuterNamedBuildTimeConfig dataSourceReactiveBuildTimeConfig = dataSourcesReactiveBuildTimeConfig
-                .dataSources().get(DataSourceUtil.DEFAULT_DATASOURCE_NAME);
+                .dataSources().get(datasourceNameFromConf);
 
         if (dataSourceReactiveBuildTimeConfig == null || !dataSourceReactiveBuildTimeConfig.reactive().enabled()) {
             LOG.warn("Hibernate Reactive is disabled because the default datasource is not reactive");
```

Comment:
It's no longer about the default datasource.

You might want to look around for similar assumptions in other logs / error messages?